cpp/src/arrow/ipc/reader.cc

```diff
@@ -2032,6 +2036,11 @@ Status StreamDecoder::Consume(const uint8_t* data, int64_t size) {
 Status StreamDecoder::Consume(std::shared_ptr<Buffer> buffer) {
   return impl_->Consume(std::move(buffer));
 }
+Status StreamDecoder::Reset() {
+  impl_ =
+      std::make_unique<StreamDecoderImpl>(std::move(impl_->listener()), impl_->options());
```

Comment:
Nit: `std::move` on a rvalue is probably a no-op.

Reply:
Oh, sorry. I'll remove it.